File: test/zeromq/node_client/suite.js
const zerorpc = require("zerorpc");

const client = new zerorpc.Client();

const port = process.env.ZEROMQ_PORT || "4242";
const host = process.env.ZEROMQ_HOST || "0.0.0.0";

console.log(`Connecting to tcp://${host}:${port}`)
client.connect(`tcp://${host}:${port}`);

client.on("error", function(error) {
  console.error("RPC client error:", error);
});

client.invoke("parse", "hotel new york", function(error, res, more) {
  if(error) {
    console.error(error);
  } else {
    console.log("Replied:", res);
  }

  if(!more) {
    console.log("Done.");
  }
});

client.invoke("nounChunks", "hotel new york", function(error, res, more) {
  if(error) {
    console.error(error);
  } else {
    console.log("Replied:", res);
  }

  if(!more) {
    console.log("Done.");
  }
});

File: test/zeromq/python_client/suite.py
import zerorpc
import json
import os

c = zerorpc.Client()

host = os.environ.get('ZEROMQ_HOST')
host = host if host else "0.0.0.0"
port = os.environ.get('ZEROMQ_PORT')
port = port if port else "4242"

conn_str = 'tcp://' + host + ':' + port
print("Conncting: " + conn_str)
c.connect(conn_str)

input = u'hotel in new york'
nounChunks = c.nounChunks(input)
print('nounChunks: ')
print(nounChunks)

actual = json.loads(nounChunks)
expected = json.loads("""[{"text": "hotel"},{"text": "new york"}]""")

print('Expected: ')
print(expected)
print('Actual: ')
print(actual)

if actual == expected:
    print('Noun chunks successful')
else:
    raise Exception('Noun chunks not same')
